qiskit_machine_learning/neural_networks/effective_dimension.py

```diff
@@ -0,0 +1,354 @@
+
+from .opflow_qnn import OpflowQNN
+from .neural_network import NeuralNetwork
+
+
+class EffectiveDimension:
+    """
+    This class computes the global effective dimension for Qiskit
+    :class:`~qiskit_machine_learning.neural_networks.NeuralNetwork`
+    following the definition used in [1].
+
+        **References**
+        [1]: Abbas et al., The power of quantum neural networks.
+        `The power of QNNs <https://arxiv.org/pdf/2011.00027.pdf>`__.
+    """
+
+    def __init__(
+        self,
+        qnn: NeuralNetwork,
+        weight_samples: Union[List[float], np.ndarray, int] = 1,
+        input_samples: Union[List[float], np.ndarray, int] = 1,
+        callback: Optional[Callable[[int, float, float], None]] = None,
+    ) -> None:
+
+        """
+        Args:
+            qnn: A Qiskit :class:`~qiskit_machine_learning.neural_networks.NeuralNetwork`,
+                with a specific dimension ``(num_weights)`` that will determine the shape of the
+                Fisher Information Matrix ``(num_input_samples * num_weight_samples, num_weights,
+                num_weights)`` used to compute the global effective dimension for a set of
+                ``input_samples``, of shape ``(num_input_samples, qnn_input_size)``, and
+                ``weight_samples``, of shape ``(num_weight_samples, num_weights)``.
+            weight_samples: An array of neural network parameters (weights), of shape
+                ``(num_weight_samples, num_weights)``, or an ``int`` to indicate the number of
+                parameter sets to sample randomly from a uniform distribution. By default,
+                ``weight_samples = 1``.
+            input_samples: An array of samples to the neural network, of shape
+                ``(num_input_samples, qnn_input_size)``, or an ``int`` to indicate the number of
+                input sets to sample randomly from a normal distribution. By default,
+                ``input_samples = 1``.
+            callback: A callback function for the Monte Carlo sampling.
+        """
+
+        # Store arguments
+        self._weight_samples = None
+        self._input_samples = None
+        self._num_weight_samples = 1
+        self._num_input_samples = 1
+        self._model = qnn
+        self._callback = callback
+
+        # Define weight samples and input samples
+        self.weight_samples = weight_samples  # type: ignore
+        # input setter uses self._model
+        self.input_samples = input_samples  # type: ignore
+
+    def num_weights(self) -> int:
+        """Returns the dimension of the model according to the definition from [1]."""
+        return self._model.num_weights
+
+    @property
+    def weight_samples(self) -> np.ndarray:
+        """Returns network parameters (weights)."""
+        return self._weight_samples
+
+    @weight_samples.setter
+    def weight_samples(self, weight_samples: Union[List[float], np.ndarray, int]) -> None:
+        """Sets network parameters (weights)."""
```

Comment:
Since the input samples setter/getter is more direct and says `Sets network input samples.` Would this and the getter above be more direct as `Sets network weights samples.` - why do we need to use the term parameters here and then have to qualify it with (weights) in order for it to presumably be understood in context here.

Reply:
Updated.